pkg/networkservice/core/chainbreak/server.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package chainbreak provides wrapper chains elements to execute Request, Close on element out of the chain
+package chainbreak
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+)
+
+// NewNetworkServiceServer wraps given chain element to execute Request, Close out of the chain
+func NewNetworkServiceServer(wrapped networkservice.NetworkServiceServer) networkservice.NetworkServiceServer {
```

Comment:
Can we simplify this element to
```suggestion
func NewNetworkServiceServer() networkservice.NetworkServiceServer {
```
?

Reply:
No, because there would be another way to use it.
Current:
```
chainbreak.NewNetworkServiceServer(wrapped).Close(ctx, conn)
```
After "simplifying":
```
next.NewNetworkServiceServer(wrapped, tail.NewNetworkServiceServer()).Close(ctx, conn)
```
So it is doesn't become better.

Reply:
Hmm... I feel the second variant is more clear. Let's consider the case where we need to break the chain of A,B,C, D, elements, where D should not be included.

Your suggestion:
```
next.NewNetworkServiceServer(chainbreak.NewNetworkServiceServer(next. NewNetworkServiceServer(A,B, C)).Close(ctx, conn), D)
```

https://github.com/networkservicemesh/sdk/pull/520#discussion_r505318824 suggestion

```
next. NewNetworkServiceServer(A,B, C, tail.NewServer(), D)
```

Reply:
`D` is not being used in such chain, why do we need it?
```go
server := next.NewNetworkServiceServer(A, B, C, tail.NewServer(), D)
```
is just the same for the:
```go
server := chainbreak.NewNetworkServiceServer(next.NewNetworkServiceServer(A, B, C))
```
and it does mean that `server.Request()` will execute only `A.Request() -> B.Request() -> C.Request()`.

Reply:
Moreover, implementing `tail.NewServer()` can provide such unhealthy chains as:
```go
server := next.NewNetworkServiceServer(A, B, C, tail.NewServer(), D)
```

Reply:
@Bolodya1997 Not quite understand, 

By this logic https://github.com/networkservicemesh/sdk/pull/520#discussion_r506059280
This chain is invalid 
```go
server := chainbreak.NewNetworkServiceServer(next.NewNetworkServiceServer(A, B, C))
```
Because you can simply use 
```go
next.NewNetworkServiceServer(A, B, C)
```

Reply:
It depends on context we use:
```go
chain := next.NewNetworkServiceServer(AA, AB, AC)
AB.Request = func(ctx, request) {
    // 1. Without chainbreak
    server := next.NewNetworkServiceServer(A, B, C)
    // 2. With chainbreak
    server := chainbreak.NewNetworkServiceServer(next.NewNetworkServiceServer(A, B, C))

    delayedRequest := request.Clone()
    time.AfterFunc(time.Second, func() {
        server.Request(ctx, delayedRequest)
    })

    return next.Server(ctx, request)
}

chain.Request()
// AA -> AB -> AC
// after 1 sec:
// 1. A -> B -> C -> AC
// 2. A -> B -> C
```

Reply:
Hm... it is more looks like we have a wrong chain. What the element brokes a chain with 1. for `timeout`?

Reply:
Issue is fixed now with closing on timeout only subsequent chain elements, with no `onTimeout` passed.